pkg/ebpftracer/c/headers/common/network.h

```diff
@@ -488,6 +488,31 @@ statfunc bool fill_tuple(struct sock *sk, tuple_t *tuple)
     return true;
 }
 
+statfunc bool fill_tuple_from_bpf_sock(struct bpf_sock *sk, tuple_t *tuple)
+{
+    tuple->family = sk->family;
+
+    switch (sk->family) {
+        case AF_INET:
+            tuple->saddr.v4addr = sk->src_ip4;
+            tuple->daddr.v4addr = sk->dst_ip4;
+
+            break;
+        case AF_INET6:
+            __builtin_memcpy(tuple->saddr.u6_addr32, sk->src_ip6, 4);
```

Comment:
Interesting, does this work without having the full socket?

Reply:
This helper just fills the tuple from sock, ensuring that it's full should be done outside. For our sock state changes it should be full anyway.